### PR TITLE
GODRIVER-1844 add comment about pinning global client to one mongos

### PR DESCRIPTION
**Pull request by iwysiu**
Some more changes, to let workload-executor be an executable instead of a test. And a comment that accidentally got left out of an earlier commit.

### CODE REVIEW

File: mongo/integration/mtest/mongotest.go [hunk 1]
```diff
@@ -685,8 +685,8 @@ func verifyServerParametersConstraints(serverParameters map[string]bson.RawValue
 	return nil
 }
 
-// verifyRunOnBlockConstraint returns an error if the current environment does not match the provided RunOnBlock.
-func verifyRunOnBlockConstraint(rob RunOnBlock) error {
+// VerifyRunOnBlockConstraint returns an error if the current environment does not match the provided RunOnBlock.
+func VerifyRunOnBlockConstraint(rob RunOnBlock) error {
```
Comment (benjirewis) [inline mongo/integration/mtest/mongotest.go:689]: Is this publicizing just an overlooked function from [exporting](https://github.com/mongodb/mongo-go-driver/commit/b7f9264bd8759ee6f922dca899daa4d4a0009ec2) the unified test runner?

File: mongo/integration/mtest/setup.go [hunk 2]
```diff
@@ -60,7 +60,8 @@ func setupClient(cs connstring.ConnString, opts *options.ClientOptions) (*mongo.
 	if opts.ServerAPIOptions == nil && testContext.requireAPIVersion {
 		opts.SetServerAPIOptions(options.ServerAPI(driver.TestServerAPIVersion))
 	}
-	// for sharded clusters, pin to one host
+	// for sharded clusters, pin to one host. Due to how the cashe is implemented on 4.0 and 4.2, behavior
```
Comment (benjirewis) [inline mongo/integration/mtest/setup.go:63]:
```suggestion
	// for sharded clusters, pin to one host. Due to how the cache is implemented on 4.0 and 4.2, behavior
```

Comment (iwysiu) [reply mongo/integration/mtest/setup.go]: done

File: mongo/integration/mtest/setup.go [hunk 3]
```diff
@@ -60,7 +60,8 @@ func setupClient(cs connstring.ConnString, opts *options.ClientOptions) (*mongo.
 	if opts.ServerAPIOptions == nil && testContext.requireAPIVersion {
 		opts.SetServerAPIOptions(options.ServerAPI(driver.TestServerAPIVersion))
 	}
-	// for sharded clusters, pin to one host
+	// for sharded clusters, pin to one host. Due to how the cache is implemented on 4.0 and 4.2, behavior
+	// can be inconsistent when multiple mongoses are used
```
Comment (kevinAlbs) [inline mongo/integration/mtest/setup.go:64]: Ah, I think I missed this in the prior PR. Nit: Is there a server ticket to link to and include in the comment?

**Review by kevinAlbs — APPROVED** [review]
LGTM with a nit

**Review by divjotarora — APPROVED** [review]

**Review by benjirewis — APPROVED** [review]

File: mongo/integration/mtest/setup.go (hunk 3 above)
Comment (iwysiu) [reply mongo/integration/mtest/setup.go:64]: Not one that I know. But considering that it got fixed but not backported in 4.4 I don't think it really matters

**Pull request merged**